src/features/swap/SwapForm.tsx

```diff
@@ -79,11 +80,30 @@ function SwapForm() {
 function SwapFormInputs({ balances }: { balances: AccountBalances }) {
   const { address, isConnected } = useAccount()
 
-  const { values, setFieldValue } = useFormikContext<SwapFormValues>()
+  const { values, errors, setFieldValue, setErrors, validateForm, setFieldTouched } =
+    useFormikContext<SwapFormValues>()
   const { amount, direction, fromTokenId, toTokenId } = values
 
+  useEffect(() => {
+    setErrors({})
+  }, [setErrors, values])
+
   const { isLoading, quote, rate } = useSwapQuote(amount, direction, fromTokenId, toTokenId)
 
+  useEffect(() => {
+    ;(async () => {
+      if (Object.keys(errors).length !== 0) {
+        await validateForm()
+      }
+    })().catch((e) => {
+      logger.error(e)
+    })
+  }, [validateForm, errors, quote])
+
+  useEffect(() => {
+    setFieldValue('quote', quote)
+  }, [quote, setFieldValue, setFieldTouched])
```

Comment:
Hmm, this is here because the form validator needs the quote into right? This seems inconsistent with the way balance info gets passed in for validation. Can you have it passed in as a param instead of added as a form value (since it's not actually a field in the form)?

Reply:
I did it this way because the quote is not accessible at the point where we call the useFormValidator. But if you know a way of doing this more nicely would be interested to know how. Also, I thought it kind of makes sense since the quote is a field in the form :)

Reply:
I slightly disagree that the quote is **field** in the form, since a field is something users input data into, but I hear you that this is a convenient way to keep the quote data bundled with the other form data. The alternative would be to keep the quote in state (either local or in the redux store), but I agree that would be more verbose. It's up to you, I'm fine with this approach :)